### DISCUSSION TITLE
Add middleware assignments to individual components to make them more granular and flexible.

### DISCUSSION THREAD

**Discussion by jameskirtland3** [livewire/livewire#6002] [3. Feature ideas]
It's possible (and likely) that there is a way to circumvent this predicament, but here is the premise:

You've got Livewire components implemented across your application, and the location of those components demands the use of distinct middleware. Sure, you can use the Livewire component to receive route requests like a Controller, but whatever middleware has been placed on that route does not extend to the route created to pass information from the browser to the server.

In my case, I'm implementing Tenancy and maintain two applications with a single Laravel install: a Central application for my team and the Tenant application used by our clients. We use middleware to initialize tenancy, but since the middleware is set in only one place for ALL livewire components, I either have to circumvent our tenancy library's initialization or circumvent Livewire's routing design.

Middleware for Laravel's routing gets as granular as the individual routes themselves, but Livewire doesn't adopt the same approach. I don't think anything necessarily needs to be changed/removed, just add the ability to assign middleware to individual components so that the responsive part of the component's lifecycle can all occur within the same middleware context as the route it was mounted in.

I've seen some "solutions" for how to do this, but it seems like Livewire components should do this natively if the docs suggest components can be used in place of controllers (as they have for multiple versions).

**Comment by joshhanley** [comment]
@jameskirtland3 yeah there is already a feature called persistent middleware, which you can use to ensure middleware (such as tenant middleware) are applied. Not sure why it's not in the V3 docs, but you can see it here in the V2 docs https://laravel-livewire.com/docs/2.x/security#persistent-middleware

Hope this helps!
